manifests/init.pp

```diff
@@ -156,20 +156,33 @@
   $shared_with_foreman_vhost = !$pulpcore_mirror
 
   $rhsm_path = '/rhsm'
-  $rhsm_port = $reverse_proxy_real ? {
-    true  => $reverse_proxy_port,
-    false => 443
-  }
+  $rhsm_port = 443
 
   ensure_packages('katello-debug')
 
   include certs::foreman_proxy
   Class['certs::foreman_proxy'] ~> Service['foreman-proxy']
 
   if $reverse_proxy_real {
-    class { 'foreman_proxy_content::reverse_proxy':
-      url  => "${foreman_url}/",
-      port => $reverse_proxy_port,
+    foreman_proxy_content::reverse_proxy { "rhsm-pulpcore-https-${reverse_proxy_port}":
```

Comment:
I'll admit I struggled mightily with naming these.

The behavior here is still to proxy `/` to Katello primary. There is nothing related to Pulpcore here, unless `$reverse_proxy_port` is overridden to 443 in which case Pulpcore attaches its configuration to this vhost instead. (in fact, that should still be tested to ensure there are no issues with the `order` of the fragments, the value of which was chosen to work with the Foreman vhost)

Reply:
I have a pipeline running with `--foreman_proxy_content_reverse_proxy_port 443` to confirm the correct functionality in that setup

Reply:
After working through various other issues with nightly builds, the outcome of these tests of 443-only was successful, and the deployment worked as expected.

In addition to bats and smoker via the nightly katello pipeline, I ran the following manual tests for verification.

1. curl pulp/api/v3/status/ and confirm that it reaches pulp running on proxy (not on katello primary):

```
# curl https://pipe-katello-proxy-nightly-centos7.mercury.example.com/pulp/api/v3/status/ 2>/dev/null | json_reformat | grep katello | head -n1
            "name": "21421@pipe-katello-proxy-nightly-centos7.mercury.example.com",
```

2. curl /rhsm/status/

```
# curl -s -o /dev/null -w "%{http_code}\n" https://$(hostname)/rhsm/status/
200
```

3. Examination of apache configs

```
# ll /etc/httpd/conf.d/
total 16
-rw-r--r--. 1 root root 2405 Nov  9 16:30 10-pulpcore.conf
-rw-r--r--. 1 root root 6262 Nov  9 16:29 10-rhsm-pulpcore-https-443.conf
-rw-r--r--. 1 root root  848 Nov  9 16:29 ssl.conf
```

4. The `10-rhsm-pulpcore-https-443.conf` config:

```ApacheConf
# ************************************
# Vhost template in module puppetlabs-apache
# Managed by Puppet
# ************************************
# 
<VirtualHost *:443>
  ServerName pipe-katello-proxy-nightly-centos7.mercury.example.com

  ## Vhost docroot
  DocumentRoot "/var/www/"

  ## Directories, there should at least be a declaration for /var/www/

  <Directory "/var/www/">
    Options Indexes FollowSymLinks MultiViews
    AllowOverride None
    Require all granted
  </Directory>

  ## Logging
  ErrorLog "/var/log/httpd/rhsm-pulpcore-https-443_error_ssl.log"
  ServerSignature Off
  CustomLog "/var/log/httpd/rhsm-pulpcore-https-443_access_ssl.log" combined 
  ErrorDocument 500 '{"displayMessage": "Internal error, contact administrator", "errors": ["Internal error, contact administrator"], "status": "500" }'
  ErrorDocument 503 '{"displayMessage": "Service unavailable or restarting, try later", "errors": ["Service unavailable or restarting, try later"], "status": "503" }'

  ## Request header rules
  ## as per http://httpd.apache.org/docs/2.2/mod/mod_headers.html#requestheader
  RequestHeader set X_RHSM_SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"

  # SSL Proxy directives
  SSLProxyEngine On
  SSLProxyCACertificateFile "/etc/pki/katello/certs/katello-default-ca.crt"
  SSLProxyMachineCertificateFile "/etc/pki/katello/private/pipe-katello-proxy-nightly-centos7.mercury.example.com-foreman-proxy-client-bundle.pem"
  ProxyPass /katello/api/v2/repositories/ https://pipe-katello-server-nightly-centos7.mercury.example.com/katello/api/v2/repositories/
  <Location /katello/api/v2/repositories/>
    ProxyPassReverse https://pipe-katello-server-nightly-centos7.mercury.example.com/
  </Location>
  SSLProxyEngine On

  ProxyPass /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/
  ProxyPassReverse /pulp_ansible/galaxy/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp_ansible/galaxy/

  <Location "/pulpcore_registry/v2/">
    RequestHeader unset REMOTE_USER
    RequestHeader set REMOTE_USER "%{SSL_CLIENT_S_DN_CN}s" env=SSL_CLIENT_S_DN_CN
    ProxyPass unix:///run/pulpcore-api.sock|http://pulpcore-api/v2/
    ProxyPassReverse unix:///run/pulpcore-api.sock|http://pulpcore-api/v2/
  </Location>

  ProxyPass /pulp/container/ unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/container/
  ProxyPassReverse /pulp/container/ unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/container/

  <Location "/pulp/deb">
    RequestHeader unset X-CLIENT-CERT
    RequestHeader set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT
    ProxyPass unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content disablereuse=on timeout=600
    ProxyPassReverse unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content
  </Location>

  <Location "/pulp/isos">
    RequestHeader unset X-CLIENT-CERT
    RequestHeader set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT
    ProxyPass unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content disablereuse=on timeout=600
    ProxyPassReverse unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content
  </Location>

  <Location "/pulp/repos">
    RequestHeader unset X-CLIENT-CERT
    RequestHeader set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT
    ProxyPass unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content disablereuse=on timeout=600
    ProxyPassReverse unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content
  </Location>
Alias /pub /var/www/html/pub

<Location /pub>
  <IfModule mod_passenger.c>
    PassengerEnabled off
  </IfModule>
  Options +FollowSymLinks +Indexes
  Require all granted
</Location>

  <Location "/pulp/content">
    RequestHeader unset X-CLIENT-CERT
    RequestHeader set X-CLIENT-CERT "%{SSL_CLIENT_CERT}s" env=SSL_CLIENT_CERT
    ProxyPass unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content disablereuse=on timeout=600
    ProxyPassReverse unix:///run/pulpcore-content.sock|http://pulpcore-content/pulp/content
  </Location>

  <Location "/pulp/api/v3">
    RequestHeader unset REMOTE_USER
    RequestHeader set REMOTE_USER "%{SSL_CLIENT_S_DN_CN}s" env=SSL_CLIENT_S_DN_CN
    RequestHeader set REMOTE_USER "admin" "expr=%{SSL_CLIENT_S_DN_CN} == 'pipe-katello-server-nightly-centos7.mercury.example.com'"
    RequestHeader set REMOTE_USER "admin" "expr=%{SSL_CLIENT_S_DN_CN} == 'pipe-katello-proxy-nightly-centos7.mercury.example.com'"
    ProxyPass unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp/api/v3 timeout=600
    ProxyPassReverse unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp/api/v3
  </Location>

  ProxyPass /pulp/assets/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp/assets/
  ProxyPassReverse /pulp/assets/ unix:///run/pulpcore-api.sock|http://pulpcore-api/pulp/assets/

  ## Proxy rules
  ProxyRequests Off
  ProxyPreserveHost Off
  ProxyPass / https://pipe-katello-server-nightly-centos7.mercury.example.com/ disablereuse=on retry=0
  ProxyPassReverse / https://pipe-katello-server-nightly-centos7.mercury.example.com/

  ## SSL directives
  SSLEngine on
  SSLCertificateFile      "/etc/pki/katello/certs/katello-apache.crt"
  SSLCertificateKeyFile   "/etc/pki/katello/private/katello-apache.key"
  SSLCertificateChainFile "/etc/pki/katello/certs/katello-server-ca.crt"
  SSLVerifyClient         optional
  SSLVerifyDepth          10
  SSLCACertificateFile    "/etc/pki/katello/certs/katello-default-ca.crt"
  SSLOptions +StdEnvVars +ExportCertData +FakeBasicAuth

  ## Container Gateway
  <Location "/pulpcore_registry/v2/">
    Require expr %{SSL_CLIENT_S_DN_CN} == "pipe-katello-proxy-nightly-centos7.mercury.example.com"
    RequestHeader set SSL_CLIENT_S_DN "admin"
  </Location>
  ProxyPass /v1/ https://pipe-katello-proxy-nightly-centos7.mercury.example.com:9090/container_gateway/v1/
  ProxyPassReverse /v1/ https://pipe-katello-proxy-nightly-centos7.mercury.example.com:9090/container_gateway/v1/
  ProxyPass /v2/ https://pipe-katello-proxy-nightly-centos7.mercury.example.com:9090/container_gateway/v2/
  ProxyPassReverse /v2/ https://pipe-katello-proxy-nightly-centos7.mercury.example.com:9090/container_gateway/v2/
</VirtualHost>
```

5. Further details of my test setup:

Changes to the foreman_installer role defaults:

```
$ egrep 'module_prs:|foreman_installer_options:' roles/foreman_installer/defaults/main.yml 
foreman_installer_module_prs: ["katello/foreman_proxy_content/390","theforeman/foreman_proxy/714"]
foreman_installer_options: ['{{ "--foreman-proxy-content-reverse-proxy-port 443" if foreman_installer_scenario != "katello" else "" }}']
```

The command used to run the pipeline:

```
ansible-playbook pipelines/install_pipeline.yml -e forklift_state=up -e pipeline_os=centos7 -e pipeline_type=katello -e pipeline_version=nightly
```